lib/session.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize,Lint/MissingCopEnableDirective
+
+module AutoHCK
+  class Session
+    extend T::Sig
+    include Helper
+
+    def self.save(workspace_path, options)
+      File.write("#{workspace_path}/session.json", compose_session_json(options, workspace_path))
+    end
+
+    def self.load(cli)
+      session = Models::Session.from_json_file("#{session_path(cli)}/session.json")
+
+      session.common.workspace_path = cli.common.workspace_path
+      session.test.latest_session = cli.test.latest_session
+      session.test.session = cli.test.session
+
+      cli.test = session.test
+      cli.common = session.common
+
+      cli.test.manual = true
+    end
+
+    def self.session_path(cli)
+      cli.test.latest_session ? "#{Config.read['workspace_path']}/latest" : cli.test.session
+    end
+
+    private_class_method def self.compose_session_json(options, workspace_path)
+      session = AutoHCK::Models::Session.new(
+        test: options.test,
+        common: options.common
+      )
+      session.common.workspace_path = workspace_path
```

Comment:
This has a side effect of modifying `options`, which is obscure and should be avoided.